src/Dropdown/Dropdown.tsx

```diff
@@ -1,400 +1,160 @@
-  disabled: t.maybe(t.Boolean),
-  searchable: t.maybe(t.Boolean),
-  clearable: t.maybe(t.Boolean),
-  backspaceRemoves: t.maybe(t.Boolean),
-  multi: t.maybe(t.Boolean),
+  isDisabled: t.maybe(t.Boolean),
+  isSearchable: t.maybe(t.Boolean),
+  isClearable: t.maybe(t.Boolean),
+  isMulti: t.maybe(t.Boolean),
   flat: t.maybe(t.Boolean),
-  autoBlur: t.maybe(t.Boolean),
-  simpleValue: t.maybe(t.Boolean),
+  blurInputOnSelect: t.maybe(t.Boolean),
+  backspaceRemovesValue: t.maybe(t.Boolean),
   menuPosition: t.maybe(t.enums.of(["top", "bottom"])),
-  menuRenderer: t.maybe(t.Function),
-  groupByKey: t.maybe(t.String),
-  optionGroupRenderer: t.maybe(t.Function),
   placeholder: t.maybe(t.union([t.String, ReactElement])),
-  noResultsText: t.maybe(t.String),
-  allowCreate: t.maybe(t.Boolean),
-  addLabelText: t.maybe(t.String),
-  valueRenderer: t.maybe(t.Function),
-  optionRenderer: t.maybe(t.Function),
+  noOptionsMessage: t.maybe(t.String),
   delimiter: t.maybe(t.String),
   onInputChange: t.maybe(t.Function),
   onFocus: t.maybe(t.Function),
   onBlur: t.maybe(t.Function),
-  onBlurResetsInput: t.maybe(t.Boolean),
-  onCloseResetsInput: t.maybe(t.Boolean),
   isLoading: t.maybe(t.Boolean),
   id: t.maybe(t.String),
   className: t.maybe(t.String),
-  style: t.maybe(t.Object)
+  style: t.maybe(t.Object),
+  components: t.maybe(t.Object)
 };
 
-export namespace Dropdown {
-  export type SimpleValue = string | number;
-
-  export type Value = SelectNS.Option | SelectNS.Options | SimpleValue;
-
-  export type OptionRendererHandler = (
-    option: SelectNS.Option,
-    i: number
-  ) => SelectNS.HandlerRendererResult;
-
-  export type OptionGroupRendererHandler = (
-    optionGroup: string
-  ) => JSX.Element | string | null;
-
-  export interface SelectMissingMenuRendererProps {
-    // seem supported by react-select but missing in the @types file?
-    instancePrefix?: string;
-    onFocus?: () => void;
-    onOptionRef: (
-      ref: React.Component<Dropdown.Props> | null,
-      isFocused: boolean
-    ) => void;
-    onSelect?: () => void;
-    optionClassName?: string;
-    optionComponent: React.ComponentType<{ [k: string]: any }>; // TODO
-    optionRenderer: OptionRendererHandler;
-    optionGroupRenderer: OptionGroupRendererHandler;
-    valueKey: string;
-  }
-  export interface CustomMenuRendererProps {
-    groupByKey?: string;
-  }
-  export type MenuRendererProps = SelectNS.MenuRendererProps &
-    SelectMissingMenuRendererProps &
-    CustomMenuRendererProps;
-  export type MenuRendererHandler = (
-    props: MenuRendererProps
-  ) => JSX.Element[] | SelectNS.HandlerRendererResult;
-
-  export type MenuPosition = "top" | "bottom";
-
-  export type Props = RequiredProps & Partial<DefaultProps>;
-}
-
-type DefaultedProps = RequiredProps & DefaultProps;
+const defaultComponents: Dropdown.Props<any>["components"] = {
+  Placeholder: ({ children }) => (
+    <div className="react-select__placeholder">{children}</div>
```

Comment:
if we are re-defining it, why not using our own new class instead of `react-select__placeholder`?

Reply:
this class is indeed not defined by `react-select` (unfortunately some elements [like placeholder] are not given a class to target them anymore).
I used a name that would follow `react-select` convention as IMHO if we use `react-select` classes it would be messy to mix in other naming conventions

Reply:
> this class is indeed not defined by `react-select` (unfortunately some elements [like placeholder] are not given a class to target them anymore).

This is not true. Those components have custom classes (https://react-select.com/styles)